File: Math/529-commonFactors.py
class Solution:
    def commonFactors(self, a: int, b: int) -> int:
        def gcd(a,b):
            while b:
                a,b=b,a%b
            return a
            if b:
                return gcd(b,a%b)
            return a
        ans=1
        x=gcd(a,b)
        for i in range (2,x+1):
            if x%i==0:
                ans+=1
        return ans

File: Math/529-commonFactors.cpp
#include<iostream>

using std::cout;

int gcd(int a,int b)
{
    while(b)
    {
        int tmp=b;
        b=a%b;
        a=tmp;
    }
    return a;
}


int commonFactors(int a,int b)
{
    int x=gcd(a,b);
    int ans=1;
    for(int i=2;i<x+1;++i)
    {
        if(x%i==0)
            ans+=1;
    }
    return ans;
}

int main()
{
    int a=12,b=24;
    cout<<commonFactors(a,b);
}